lib/qesapdeployment.pm

```diff
@@ -1786,12 +1786,20 @@ sub qesap_is_job_finished {
 
     my $job_data = eval { decode_json($json_data) };
     if ($@) {
-        record_info("JSON error", "Failed to decode JSON data for job $job_id: $@");
-        return 0;    # Assume job is still running if we can't get its state
+        if ($json_data =~ /<h1>Page not found<\/h1>/) {
```

Comment:
you can do it even immediately after line 1785, before the `decode_json`

Reply:
Yes, but I grouped them so I can use the same return statement for both (unhealthy obsession with code reduction, I know)